components/3d/components/three/LoadingScreen.tsx

```diff
@@ -0,0 +1,58 @@
+import { Html, useProgress } from '@react-three/drei';
+import useGlobalStore from '../../stores/useGlobalStore';
+import { useEffect, useState } from 'react';
+import ProgressBar from '@ramonak/react-progress-bar';
+
+export default function LoadingScreen() {
+  //Wrapper hook of THREE.DefaultLoadingManager progress status
+  const progress = useProgress((state) => state.progress);
+
+  // - STATE
+  //useProgress can give smaller numbers than current progress due to bug, we will only fetch when it increases (see useEffect)
+  const [realProgress, setRealProgress] = useState(15); //UX: its much better if the user don't see a 0%
+
+  // - STATE GLOBAL ZUSTAND
+  const { set3DLoaded } = useGlobalStore((state) => {
+    return {
+      set3DLoaded: state.set3DLoaded,
+    };
+  });
+
+  // - EFFECT
+  useEffect(() => {
+    //unmount - LoadingScreen is unmounted when the linked <Suspense fallback> is finished
+    return () => {
+      set3DLoaded(true);
```

Comment:
Is quite strange performing the `set3DLoaded(true);` when LoadingScreen is unmounted.
What should be done is unmount the LoadingScreen when is3DLoaded is true, and for that, LoadingScreen should not be responsible to set the 3D loaded state. I don't know if this is possible with the Suspense and all of this, but maybe there is a Suspense callbak onLoad or something and you can set3dLoaded there... but just saying.

Is not critical BTW, but i just wanted to comment it.

Reply:
The unmount un React comes from the return callback function of useEffect().

When using <Suspense> the best way to know the moment that the <Suspense> finishes is by executing code inside the unmount (useEffect return allback function) of the fallback prop component. 

A future improvement would be to convert this inline return cb to a "unmountCB" prop of LoadingScreen, allowing the parent component to handle what happens when the <Suspense> finishes their job.

For the moment, I will let the code stay here because I think it makes more understandable how this specific flow works.

Reply:
You can leave this as it is but IMO this is a risky workaround as you are always assuming that the suspense is only wrapping the map loading and no other async tasks that can occur inside.
I don't know exactly the async tasks running inside that make the suspense to appear and how critical is to update the "isLoaded" state, but this should be handled in the component running the async tasks. Maybe there is a reason that I'm missing

Reply:
@sergiogc9 @llorenspujol after a short revision, I just saw how much more natural it is to handle it by the parent component (the one that manages the async tasks), following both your opinions.

Also, it removes from the repo a global store used only for this function, in a way, highlighting how this data binding was poorly done! https://github.com/gironajs/gironajs.com/pull/6/commits/afc8cb2151fcc804bcdba3baaa1b5ba1ce6a2ef2